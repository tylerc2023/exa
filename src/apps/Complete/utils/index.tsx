export * from './configs';
export * from './useKeyboardHeight';
export * from './userFactory';
