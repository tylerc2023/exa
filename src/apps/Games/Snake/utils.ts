export type BoardObject = Array<Array<0 | 1 | 2>>;

export const generateRandom = (size: number) =>
  Math.floor(Math.random() * size);

export const addFood = (
  board: BoardObject,
  used: {[key: string]: boolean} = {},
): boolean => {
  const x = generateRandom(board.length);
  const y = generateRandom(board.length);

  const combo = `${x}${y}`;
  if (Object.keys(used).length === board.length) {
    return false;
  }
  if (combo in used || board[x][y] !== 0) {
    used[combo] = true;
    addFood(board, used);
    return false;
  }
  board[y][x] = 2;
  return true;
};

export const addStarting = (board: BoardObject) => {
  const center = Math.floor(board.length / 2);
  board[center][center] = 1;
};

export const generateBoard = (size: number) => {
  const board: BoardObject = [];
  for (let i = 0; i < size; i++) {
    board[i] = [];
    for (let j = 0; j < size; j++) {
      board[i][j] = 0;
    }
  }

  addStarting(board);
  addFood(board);
  return board;
};

export const nextSnakePosition = () => undefined;
export const collision = (next: any) => next && false;
export const endGame = () => undefined;
export const eat = () => undefined;
