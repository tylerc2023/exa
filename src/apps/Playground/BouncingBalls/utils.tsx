import {Animated} from 'react-native';
import {State} from './BouncingBall';
import {CanvasDimensions} from './Canvas';

const getRandomNumber = (min: number, max: number): number =>
  Math.random() * (max - min) + min;

type GetSpeedProps = {
  state: State;
  canvas: CanvasDimensions;
  diameter: number;
  pulse: boolean;
};

export const getSpeed = ({state, canvas, diameter, pulse}: GetSpeedProps) => {
  const dMax = state.diameter >= diameter * 1.1;
  const dMin = state.diameter <= diameter * 0.9;
  // TODO: recenter so pulsing happens from middle
  const growing = dMax || dMin ? !state.growing : state.growing;
  const d = state.diameter + (!pulse ? 0 : growing ? 1 : -1);
  const xMin = canvas.x;
  const xMax = canvas.x + canvas.width;
  const yMin = canvas.y;
  const yMax = canvas.y + canvas.height;
  const xLarger = state.x + d >= xMax;
  const xSmaller = state.x <= xMin;
  const yLarger = state.y + d >= yMax;
  const ySmaller = state.y <= yMin;
  // TODO: add collision
  const xSpeed = xLarger || xSmaller ? -state.xSpeed : state.xSpeed;
  const ySpeed = yLarger || ySmaller ? -state.ySpeed : state.ySpeed;
  const x = xLarger ? xMax - d : xSmaller ? xMin : state.x + xSpeed;
  const y = yLarger ? yMax - d : ySmaller ? yMin : state.y + ySpeed;
  const dx = x + xSpeed;
  const dy = y + ySpeed;
  return {xSpeed, ySpeed, x: dx, y: dy, d, growing};
};

type GetInitialStateProps = {
  canvas: CanvasDimensions;
  diameter: number;
  speed: number;
};

export const getInitialState = ({
  canvas,
  diameter,
  speed,
}: GetInitialStateProps): State => {
  const starting = {
    x: getRandomNumber(canvas.x, canvas.x + canvas.width - diameter),
    y: getRandomNumber(canvas.y, canvas.y + canvas.height - diameter),
  };
  const growing = !!Math.floor(Math.random());
  return {
    ...starting,
    position: new Animated.ValueXY(starting),
    xSpeed: getRandomNumber(-speed, speed),
    ySpeed: getRandomNumber(-speed, speed),
    diameter,
    growing,
  };
};
