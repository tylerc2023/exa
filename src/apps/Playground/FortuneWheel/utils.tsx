type GetRandomRange = {
  min: number;
  max: number;
};

const getRandomRange = ({min, max}: GetRandomRange) => {
  return Math.floor(Math.random() * (max - min) + min);
};

type GetNewLocation = {
  minSpin: number;
  maxSpin: number;
  numOfSegments: number;
  location: number;
};

export const getNewLocation = ({
  minSpin,
  maxSpin,
  numOfSegments,
  location,
}: GetNewLocation) => {
  const spins = getRandomRange({min: minSpin, max: maxSpin}) * 360;
  const winner = Math.floor(Math.random() * numOfSegments);
  const winnerSection = Math.floor(winner * (360 / numOfSegments));
  const newLocation = location + spins + winnerSection;
  return newLocation;
};

type GetWinnerIndex = {
  location: number;
  numOfSegments: number;
  angleOfSegment: number;
};

export const getWinnerIndex = ({
  location,
  numOfSegments,
  angleOfSegment,
}: GetWinnerIndex) => {
  const deg = Math.abs(Math.round(location % 360));
  const index =
    (numOfSegments - Math.floor(deg / angleOfSegment)) % numOfSegments;
  return index;
};
