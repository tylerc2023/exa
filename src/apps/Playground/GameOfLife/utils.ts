const operations = [
  [0, 1],
  [0, -1],
  [1, -1],
  [-1, 1],
  [1, 1],
  [-1, -1],
  [1, 0],
  [-1, 0],
];

type BoardItem = 0 | 1;

export type Board = Array<Array<BoardItem>>;

export const generateBoard = (size: number, random = 0): Board => {
  const board: Board = [];
  for (let i = 0; i < size; i++) {
    board[i] = [];
    for (let j = 0; j < size; j++) board[i][j] = Math.random() > random ? 0 : 1;
  }
  return board;
};

export const determineBoardItem = (
  board: Board,
  i: number,
  j: number,
): BoardItem => {
  let neighbors = 0;
  const item = board[i][j];
  const length = board.length;
  for (let k = 0; k < operations.length; k++) {
    const [x, y] = operations[k];
    const newI = i + x;
    const newK = j + y;
    if (newI >= 0 && newI < length && newK >= 0 && newK < length)
      neighbors += board[newI][newK];
    if (neighbors > 3) break;
  }

  return neighbors < 2 || neighbors > 3
    ? 0
    : item === 0 && neighbors === 3
    ? 1
    : item;
};
