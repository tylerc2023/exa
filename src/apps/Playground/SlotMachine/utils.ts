import {Combinations, Reel, Reels} from './config';

const getWinsPerCombination = (
  combinations: Combinations,
  reels: Reels,
): Combinations => {
  const output = [...combinations];
  for (let i = 0; i < combinations.length; i++) {
    let freq = 1;
    output[i] = [...combinations[i]];
    for (let j = 0; j < reels.length; j++) {
      const symbol = output[i][j];
      const reel = reels[j];
      const sum = getSumOfReel(reel);
      freq *=
        symbol !== 'any' && typeof symbol !== 'number' ? reel[symbol] : sum;
    }
    output[i][3] = freq;
  }
  return output;
};

const getSumOfReel = (reel: Reel): number => {
  return Object.values(reel).reduce((sum, item) => sum + item, 0);
};

const getSumOfReels = (reels: Reels): number => {
  return reels.reduce((total, reel) => total * getSumOfReel(reel), 1);
};

const getReturnsPerCombination = (
  combinations: Combinations,
  winsPerCombination: Combinations,
): Combinations => {
  return winsPerCombination.reduce(
    (output: Combinations, combination, index) => {
      const payout = combinations[index][3];
      output[index][3] = combination[3] * payout;
      return output;
    },
    [...winsPerCombination],
  );
};

const getSumOfCombinations = (combinations: Combinations): number => {
  return combinations.reduce((total, combination) => total + combination[3], 0);
};

export const getWinPercentage = (
  combinations: Combinations,
  reels: Reels,
): number => {
  const sumOfReels = getSumOfReels(reels);
  const winsPerCombination = getWinsPerCombination(combinations, reels);
  const sumOfWins = getSumOfCombinations(winsPerCombination);
  return sumOfWins / sumOfReels;
};

export const getReturnPercentage = (
  combinations: Combinations,
  reels: Reels,
): number => {
  const sumOfReels = getSumOfReels(reels);
  const winsPerCombination = getWinsPerCombination(combinations, reels);
  const returnsPerCombination = getReturnsPerCombination(
    combinations,
    winsPerCombination,
  );
  const sumOfReturns = getSumOfCombinations(returnsPerCombination);
  return sumOfReturns / sumOfReels;
};
