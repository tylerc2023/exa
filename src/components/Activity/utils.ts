import axios from 'axios';
import endOfWeek from 'date-fns/endOfWeek';
import format from 'date-fns/format';
import isToday from 'date-fns/isToday';
import startOfWeek from 'date-fns/startOfWeek';
import sub from 'date-fns/sub';
import {ActivityMatrix, Site} from './types';

type ApiResponse = {[unix: string]: number};

type ApiPromise = Promise<ApiResponse>;

type ApiInput = {
  username: string;
  site: Site;
};

type ActivitySquares = {
  matrix: ActivityMatrix;
  max: number;
  total: number;
};

export const getDateFormat = (date: number): string =>
  format(date, 'yyyy-MM-dd');

export const getSubmissionFormat = (count: number, date: number): string => {
  const submissions = count === 1 ? 'submission' : 'submissions';
  const day = isToday(date) ? 'today' : `on ${format(date, 'MMM dd, yyyy')}`;
  return `${count} ${submissions} ${day}`;
};

export const getActivitySquares = (): ActivitySquares => {
  const today = Date.now();
  const matrix = [];
  const oneDay = 60 * 60 * 24 * 1000;
  const begin = sub(startOfWeek(today), {years: 1}).valueOf();
  let end = endOfWeek(today).valueOf();
  let day = 0;
  let week = [];
  while (end >= begin) {
    week.unshift({date: end, count: 0});
    if (day > 5) {
      matrix.push(week);
      week = [];
      day = 0;
    } else {
      day++;
    }
    end -= oneDay;
  }

  return {matrix, max: 0, total: 0};
};

export const updateActivitySquares = (
  squares: ActivitySquares,
  active: ApiResponse,
): ActivitySquares => {
  let max = 0;
  let total = 0;
  const matrix = squares.matrix.map((week) =>
    week.map((day) => {
      const count = active[getDateFormat(day.date)] || 0;
      max = Math.max(count, max);
      total += count;
      return {...day, count};
    }),
  );
  return {matrix, max, total};
};

const getGithubActivity = async (username: string): Promise<ApiResponse> => {
  const url = `https://github-contributions-json-api.now.sh/api?username=${username}`;
  const res = await axios.get(url);
  return res.data;
};

const getLeetCodeActivity = async (username: string) => {
  const url = `https://leetcode.com/api/user_submission_calendar/${username}`;
  const res = await axios.get(url);
  const data = JSON.parse(res.data);
  return Object.keys(data).reduce((total: ApiResponse, item) => {
    const day = getDateFormat(Number(item) * 1000);
    if (day in total) total[day] += data[item];
    else total[day] = data[item];

    return total;
  }, {});
};

const getHackerRankActivity = async (username: string) => {
  const url = `https://www.hackerrank.com/rest/hackers/${username}/submission_histories`;
  const res = await axios.get(url);
  return Object.keys(res.data).reduce((total: ApiResponse, item) => {
    total[item] = Number(res.data[item]);
    return total;
  }, {});
};

export const getApiActivity = ({username, site}: ApiInput): ApiPromise => {
  switch (site) {
    case 'github':
      return getGithubActivity(username);
    case 'leetCode':
      return getLeetCodeActivity(username);
    case 'hackerRank':
      return getHackerRankActivity(username);
    default:
      return Promise.resolve({});
  }
};
