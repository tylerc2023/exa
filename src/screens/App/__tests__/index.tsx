import * as React from "react";
import { create } from "react-test-renderer";
import { App } from "..";
import { MockProviderAndRouter } from "../../../utils";

it("renders correctly", () => {
  const dom = create(<MockProviderAndRouter Component={App} />).toJSON();
  expect(dom).toMatchSnapshot();
});
