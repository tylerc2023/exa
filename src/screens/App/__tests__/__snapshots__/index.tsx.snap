// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "hsl(0, 100%, 100%)",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
          "padding": 12,
        }
      }
    >
      <View
        style={
          Array [
            undefined,
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "hsl(210, 10%, 23%)",
                "fontSize": 24,
              },
              Array [
                undefined,
                undefined,
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ÔÅç
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "hsl(0, 100%, 100%)",
          "flex": 1,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <RCTScrollView
        data={
          Array [
            Object {
              "archive": false,
              "createdAt": 1555243588884,
              "createdBy": "User steve",
              "id": "6",
              "message": "again",
              "updatedAt": 1555258228887,
            },
            Object {
              "archive": false,
              "createdAt": 1555168228887,
              "createdBy": "User bob",
              "id": "5",
              "message": "another message",
              "updatedAt": 1555258228888,
            },
            Object {
              "archive": false,
              "createdAt": 1555056628889,
              "createdBy": "User steve",
              "id": "4",
              "message": "first message",
              "updatedAt": 1555258228889,
            },
            Object {
              "archive": false,
              "createdAt": 1555063828889,
              "createdBy": "User bob",
              "id": "3",
              "message": "third message",
              "updatedAt": 1555258228889,
            },
            Object {
              "archive": false,
              "createdAt": 1554991828889,
              "createdBy": "User bob",
              "id": "2",
              "message": "second message",
              "updatedAt": 1555258228889,
            },
            Object {
              "archive": false,
              "createdAt": 1554995428889,
              "createdBy": "User bob",
              "id": "1",
              "message": "helloasdinaso dinasdoias daso dinaso dna nasd oiasnd ona niosao ao noinoas ao nodasno d",
              "updatedAt": 1555258228889,
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        invertStickyHeaders={true}
        inverted={true}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Array [
            Object {
              "transform": Array [
                Object {
                  "scaleY": -1,
                },
              ],
            },
            undefined,
          ]
        }
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "column-reverse",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "width": 40,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/assets/line-chart.png",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 20,
                      "paddingTop": 30,
                      "resizeMode": "contain",
                      "width": 20,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flexShrink": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 20,
                          "fontWeight": "600",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        Object {
                          "fontWeight": "600",
                        },
                        Object {
                          "paddingRight": 8,
                        },
                      ]
                    }
                  >
                    User steve
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                          "fontWeight": "300",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    5:06 AM
                  </Text>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 16,
                        "fontWeight": "300",
                        "letterSpacing": 0.5,
                      },
                      undefined,
                      undefined,
                      Object {
                        "paddingTop": 5,
                      },
                    ]
                  }
                >
                  again
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "column-reverse",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "width": 40,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/assets/line-chart.png",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 20,
                      "paddingTop": 30,
                      "resizeMode": "contain",
                      "width": 20,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flexShrink": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 20,
                          "fontWeight": "600",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        Object {
                          "fontWeight": "600",
                        },
                        Object {
                          "paddingRight": 8,
                        },
                      ]
                    }
                  >
                    User bob
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                          "fontWeight": "300",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    8:10 AM
                  </Text>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 16,
                        "fontWeight": "300",
                        "letterSpacing": 0.5,
                      },
                      undefined,
                      undefined,
                      Object {
                        "paddingTop": 5,
                      },
                    ]
                  }
                >
                  another message
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "column-reverse",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "width": 40,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/assets/line-chart.png",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 20,
                      "paddingTop": 30,
                      "resizeMode": "contain",
                      "width": 20,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flexShrink": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 20,
                          "fontWeight": "600",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        Object {
                          "fontWeight": "600",
                        },
                        Object {
                          "paddingRight": 8,
                        },
                      ]
                    }
                  >
                    User steve
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                          "fontWeight": "300",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    1:10 AM
                  </Text>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 16,
                        "fontWeight": "300",
                        "letterSpacing": 0.5,
                      },
                      undefined,
                      undefined,
                      Object {
                        "paddingTop": 5,
                      },
                    ]
                  }
                >
                  first message
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "column-reverse",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "width": 40,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/assets/line-chart.png",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 20,
                      "paddingTop": 30,
                      "resizeMode": "contain",
                      "width": 20,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flexShrink": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 20,
                          "fontWeight": "600",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        Object {
                          "fontWeight": "600",
                        },
                        Object {
                          "paddingRight": 8,
                        },
                      ]
                    }
                  >
                    User bob
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                          "fontWeight": "300",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    3:10 AM
                  </Text>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 16,
                        "fontWeight": "300",
                        "letterSpacing": 0.5,
                      },
                      undefined,
                      undefined,
                      Object {
                        "paddingTop": 5,
                      },
                    ]
                  }
                >
                  third message
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "column-reverse",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "width": 40,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/assets/line-chart.png",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 20,
                      "paddingTop": 30,
                      "resizeMode": "contain",
                      "width": 20,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flexShrink": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 20,
                          "fontWeight": "600",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        Object {
                          "fontWeight": "600",
                        },
                        Object {
                          "paddingRight": 8,
                        },
                      ]
                    }
                  >
                    User bob
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                          "fontWeight": "300",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    7:10 AM
                  </Text>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 16,
                        "fontWeight": "300",
                        "letterSpacing": 0.5,
                      },
                      undefined,
                      undefined,
                      Object {
                        "paddingTop": 5,
                      },
                    ]
                  }
                >
                  second message
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "column-reverse",
                },
                Object {
                  "transform": Array [
                    Object {
                      "scaleY": -1,
                    },
                  ],
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "width": 40,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/assets/line-chart.png",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 20,
                      "paddingTop": 30,
                      "resizeMode": "contain",
                      "width": 20,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "flexShrink": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 20,
                          "fontWeight": "600",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        Object {
                          "fontWeight": "600",
                        },
                        Object {
                          "paddingRight": 8,
                        },
                      ]
                    }
                  >
                    User bob
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                          "fontWeight": "300",
                          "letterSpacing": 0.15,
                        },
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    8:10 AM
                  </Text>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 16,
                        "fontWeight": "300",
                        "letterSpacing": 0.5,
                      },
                      undefined,
                      undefined,
                      Object {
                        "paddingTop": 5,
                      },
                    ]
                  }
                >
                  helloasdinaso dinasdoias daso dinaso dna nasd oiasnd ona niosao ao noinoas ao nodasno d
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              blurOnSubmit={true}
              disableFullscreenUI={true}
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor="hsl(208, 7%, 46%)"
              rejectResponderTermination={true}
              selectionColor="hsl(211, 100%, 50%)"
              style={
                Array [
                  Object {
                    "backgroundColor": "hsl(0, 100%, 100%)",
                    "borderColor": "hsl(208, 7%, 46%)",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "padding": 8,
                    "paddingRight": 32,
                    "width": "100%",
                  },
                  "",
                  false,
                  Object {
                    "fontSize": 14,
                    "fontWeight": "300",
                    "letterSpacing": 0.25,
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "hsl(211, 100%, 50%)",
                "flexDirection": "row",
                "height": 20,
                "justifyContent": "flex-start",
                "marginVertical": 4,
                "opacity": 1,
                "paddingHorizontal": 0,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "opacity": 0,
                  },
                  Array [
                    "",
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "hsl(211, 100%, 50%)",
                      },
                      undefined,
                      undefined,
                      Object {
                        "color": "hsl(354, 70%, 54%)",
                      },
                      undefined,
                      undefined,
                      undefined,
                    ],
                  ],
                ]
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "hsl(210, 10%, 23%)",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "opacity": 0,
                      },
                      Array [
                        "",
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "hsl(211, 100%, 50%)",
                          },
                          undefined,
                          undefined,
                          Object {
                            "color": "hsl(354, 70%, 54%)",
                          },
                          undefined,
                          undefined,
                          undefined,
                        ],
                      ],
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ÔÄ®
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "hsl(211, 100%, 50%)",
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <View
            style={
              Array [
                undefined,
                Array [
                  undefined,
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "hsl(211, 100%, 50%)",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ],
                ],
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "hsl(210, 10%, 23%)",
                    "fontSize": 16,
                  },
                  Array [
                    undefined,
                    Array [
                      undefined,
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "hsl(211, 100%, 50%)",
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ],
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              Ôíä
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
